Orchestrator.py

```diff
@@ -42,14 +42,19 @@ def __init__(self, numRobots, initX, initY):
             (
                 i,
                 {
-                   'x':       initX,
-                   'y':       initY,
-                   'heading': 0,
-                   'speed':   0,
+                   'x':                initX,
+                   'y':                initY,
+                   'heading':          0,
+                   'speed':            0,
+                   'movementDuration': 0
```

Comment:
you don't need to keep this in memory. The DotBot indicates how long it has moved; just use the info to update the DotBot's position at reception, no need to remember.